frontend/src/app/main/ui/workspace/sidebar/layers.cljs

```diff
@@ -84,101 +84,114 @@
        (:name shape "")
        (when (seq (:touched shape)) " *")])))
 
-(defn- make-collapsed-iref
-  [id]
-  #(-> (l/in [:expanded id])
-       (l/derived refs/workspace-local)))
-
 (mf/defc layer-item
   [{:keys [index item selected objects] :as props}]
   (let [id         (:id item)
-        selected?  (contains? selected id)
-        container? (or (cph/frame-shape? item)
-                       (cph/group-shape? item))
+        blocked?   (:blocked item)
+        hidden?    (:hidden item)
 
         disable-drag      (mf/use-state false)
         scroll-to-middle? (mf/use-var true)
+        expanded-iref     (mf/with-memo [id]
+                            (-> (l/in [:expanded id])
+                                (l/derived refs/workspace-local)))
 
-        expanded-iref (mf/use-memo
-                       (mf/deps id)
-                       (make-collapsed-iref id))
-
-        expanded? (mf/deref expanded-iref)
+        expanded?         (mf/deref expanded-iref)
+        selected?         (contains? selected id)
+        container?        (or (cph/frame-shape? item)
+                              (cph/group-shape? item))
 
         toggle-collapse
-        (fn [event]
-          (dom/stop-propagation event)
-          (if (and expanded? (kbd/shift? event))
-            (st/emit! (dwc/collapse-all))
-            (st/emit! (dwc/toggle-collapse id))))
+        (mf/use-fn
+         (mf/deps expanded?)
+         (fn [event]
```

Comment:
Is it equivalent to
```
(mf/with-memo [expanded?]
  (fn [event]
    ...
```
?

Reply:
Since useCallback(fn, deps) is equivalent to useMemo(() => fn, deps)
[source](https://hyp.is/1RVjig8GEe2WoHvn4o74fg/reactjs.org/docs/hooks-reference.html)

Reply:
Yep, in fact the `use-fn` (the React.useCallback) is a specialization of React.useMemo (penpot's mf/use-memo or mf/with-memo macro)